plugins/dependencies/src/main/kotlin/org/gradle/kotlin/dsl/IsNotADependency.kt

```diff
@@ -0,0 +1,29 @@
+@file:Suppress("DeprecatedCallableAddReplaceWith")
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.dsl.DependencyHandler
+
+/**
+ * Marker interface that prevents the user to add a wrong dependency like
+ *
+ *     implementation(Testing.kotest)
+ *                   ^^^^^^^^^^^^^^^
+ *                   IDE: This is not a valid dependency notation but a group of dependency notations
+ */
+interface IsNotADependency
+
+const val ErrorIsNotADependency = """This is not a valid dependency notation but a group of dependency notations"""
+
+@kotlin.Deprecated(ErrorIsNotADependency)
+fun DependencyHandler.`implementation`(dependencyNotation: IsNotADependency): Dependency? =
+    null
+
+@kotlin.Deprecated(ErrorIsNotADependency)
+fun DependencyHandler.`api`(dependencyNotation: IsNotADependency): Dependency? =
+    null
+
+@kotlin.Deprecated(ErrorIsNotADependency)
+fun DependencyHandler.`testImplementation`(dependencyNotation: IsNotADependency): Dependency? =
+    null
```

Comment:
We can add other configurations (`compileOnly`, `runtimeOnly`& `classpath`), plus their overloads taking a lambda (though it's not fully supported for rich `version { … }` by refreshVersions yet).

Reply:
Le mieux est l'ennemi du bien :)

Reply:
@jmfayard Why not add it, and why would it be the enemy of good in this case?